Automattic/wp-calypso
# chore: upgrade `react-query` (v3) to `@tanstack/react-query` (v4)

**Second attempt**. We had to revert the first attempt (#74381) due to some regressions found.

## Proposed Changes

* Upgrade the `react-query@3` package to `@tanstack/react-query@4`
* Include any adjustments necessary because of breaking changes

## Testing Instructions

[Overview of changes between v3 and v4.](https://tanstack.com/query/latest/docs/react/guides/migrating-to-react-query-4)

**Please note**, this PR touches almost all queries and mutations for which we use the `react-query` library (now `@tanstack/react-query`) as well as a few other areas in the Calypso codebase (except for a few `/apps` directories which are not imported or don't import related code from Calypso or other packages.

* Verify Calypso builds well and does not contain any new type errors or warnings
* Depending on the query or mutation, verify data is fetched as expected (compare with prod) and no errors are thrown during runtime
* Verify the local persistence plugin still works as expected

**Additional steps** to verify regressions found during the first attempt of merging the update don't occur anymore:

- Verify the HelpCenter is working correctly & articles are shown
- Verify the signup flows works correctly (a failing pre-release e2e test showed there was an error when selecting a category the new site will be about at `/setup/site-setup/vertical`)

## Pre-merge Checklist
